File: day2/part1.py
def is_double_sequence(n: int) -> bool:
    s = str(n)
    if len(s) % 2 != 0:
        return False
    half = len(s) // 2
    return s[:half] == s[half:]



total_sum = 0

with open("in.txt", "r") as fin, open("out.txt", "w") as fout:
    with open("in.txt") as f:
        lines = f.readline().strip()
    lines = lines.split(",")
    for line in lines: 
        parts = line.split("-")
        if len(parts) < 2:
            continue

        a = int(parts[0])
        b = int(parts[1])

        fout.write(f"{a} {b}\n")

        for i in range(a, b + 1):
            if is_double_sequence(i):
                total_sum += i

    fout.write(f"result:{total_sum}\n")

File: day2/part1.cpp
#include <bits/stdc++.h>
using namespace std;

bool isDoubleSequence(long long n) {
    string s = to_string(n);
    if (s.length() % 2 != 0) return false;
    int half = s.length() / 2;
    return s.substr(0, half) == s.substr(half);
}

int main() {
    ifstream in("in.txt");
    ofstream out("out.txt");

    string line;
    getline(in, line);

    long long total = 0;

    stringstream ss(line);
    string part;

    while (getline(ss, part, ',')) {
        size_t dash = part.find('-');
        if (dash == string::npos) continue;

        long long a = stoll(part.substr(0, dash));
        long long b = stoll(part.substr(dash + 1));

        out << a << " " << b << "\n";

        for (long long i = a; i <= b; i++) {
            if (isDoubleSequence(i)) {
                total += i;
            }
        }
    }

    out << "result:" << total;
}
